src/zmalloc.c

```diff
@@ -458,8 +456,10 @@ void *zrealloc_dram(void *ptr, size_t size) {
     newptr = realloc_dram(ptr,size);
```

Comment:
You don't use oldsize variable in case of USE_MEMKIND_MEMTIER consider this in ifdef block

Reply:
Done.